pkg/cluster/util.go

```diff
@@ -386,3 +388,32 @@ func (c *Cluster) credentialSecretNameForCluster(username string, clusterName st
 func masterCandidate(replicas []spec.NamespacedName) spec.NamespacedName {
 	return replicas[rand.Intn(len(replicas))]
 }
+
+func cloneSpec(from *spec.Postgresql) (*spec.Postgresql, error) {
+	var (
+		buf    bytes.Buffer
+		result *spec.Postgresql
+		err    error
+	)
+	enc := gob.NewEncoder(&buf)
+	if err = enc.Encode(*from); err != nil {
```

Comment:
The `gob` package [explicitly prohibits `nil` pointers](https://golang.org/pkg/encoding/gob/#hdr-Basics) in the object to encode. Are all fields of the spec guaranteed to be non-`nil` ?

Reply:
yes, we don't have any explicit pointers in the spec, and empty maps or slices are accepted by gob.